File: LeetCode/MaximumSubarray/MaximumSubarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int MaxSum = nums[0];
        int Sum = 0;
        if(nums.Length == 1){
            return nums[0];
        }

        for(int i = 0; i<nums.Length; i++){
            Sum += nums[i];
            MaxSum = Math.Max(Sum, MaxSum); 

            if(Sum<0){
                Sum = 0;
            }
        }

        return MaxSum;
    }
}

File: LeetCode/MaximumSubarray/MaximumSubarray.ts
function maxSubArray(nums: number[]): number {
  let maxi = Number.NEGATIVE_INFINITY;
  let sum = 0;

  if (nums.length < 2) {
    return nums[0];
  }

  for (let i = 0; i < nums.length; i++) {
    sum += nums[i];
    if (sum > maxi) {
      maxi = sum;
    }
    if (sum < 0) {
      sum = 0;
    }
  }

  return maxi;
}
